source/agora/utils/PrettyPrinter.d

```diff
@@ -411,6 +413,70 @@ GCOQ...LRIJ(61,000,000), GCOQ...LRIJ(61,000,000)`;
     assert(ResultStr == actual, actual);
 }
 
+/// Format array of blocks
+private struct BlocksFmt
+{
+    private const(Block[]) value;
+
+    public this (ref const Block[] r) @safe nothrow
+    {
+        this.value = r;
+    }
+
+    public void toString (scope void delegate (in char[]) @safe sink)
+        @safe nothrow
+    {
+        try
+        {
+            string sep1 = "\n===============================================================================\n";
+            formattedWrite(sink, "Blocks:%s", sep1);
```

Comment:
It's really unnecessary to go through a format call when we can just do:

```D
sink("Blocks:\n===============================================================================\n")
```

Reply:
Updated